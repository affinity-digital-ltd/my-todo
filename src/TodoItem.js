import React, { Component } from 'react'

import './scss/todo-item.scss'

class TodoItem extends Component {
  constructor () {
    super()

    this.state = {
      done: 'c-todo-item__content'
    }

    this.markAsDone = this.markAsDone.bind(this)
  }

  markAsDone (event) {
    const classes = event.target.checked ? 'c-todo-item__content u-strikethrough' : 'c-todo-item__content'

    this.setState({
      done: classes
    })
  }

  render () {
    const { done } = this.state
    const { item, deleteItem } = this.props

    return (
      <div className='c-todo-item'>
        <label className='c-todo-item__task'>
          <input type='checkbox' onClick={this.markAsDone} /><span className={done}>{item}</span>
        </label>
        <button onClick={deleteItem} className='c-todo-item__delete' data-behavior='delete'>Delete</button>
      </div>
    )
  }
}

export default TodoItem
