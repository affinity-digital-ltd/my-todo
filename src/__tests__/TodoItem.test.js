import React from 'react'
import TodoItem from '../TodoItem'

import { shallow } from 'enzyme'

describe('<TodoItem />', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<TodoItem />)

    expect(wrapper).toMatchSnapshot()
  })
})