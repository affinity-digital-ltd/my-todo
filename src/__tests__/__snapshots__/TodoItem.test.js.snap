// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoItem /> should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodoItem />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          className="c-todo-item__task"
        >
          <input
            onClick={[Function]}
            type="checkbox"
          />
          <span
            className="c-todo-item__content"
          />
        </label>,
        <button
          className="c-todo-item__delete"
          data-behavior="delete"
        >
          Delete
        </button>,
      ],
      "className": "c-todo-item",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              onClick={[Function]}
              type="checkbox"
            />,
            <span
              className="c-todo-item__content"
            />,
          ],
          "className": "c-todo-item__task",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onClick": [Function],
              "type": "checkbox",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "c-todo-item__content",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Delete",
          "className": "c-todo-item__delete",
          "data-behavior": "delete",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "Delete",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            className="c-todo-item__task"
          >
            <input
              onClick={[Function]}
              type="checkbox"
            />
            <span
              className="c-todo-item__content"
            />
          </label>,
          <button
            className="c-todo-item__delete"
            data-behavior="delete"
          >
            Delete
          </button>,
        ],
        "className": "c-todo-item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                onClick={[Function]}
                type="checkbox"
              />,
              <span
                className="c-todo-item__content"
              />,
            ],
            "className": "c-todo-item__task",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "onClick": [Function],
                "type": "checkbox",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "c-todo-item__content",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Delete",
            "className": "c-todo-item__delete",
            "data-behavior": "delete",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Delete",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
